MaterialDesignThemes.Wpf/PaletteHelper.cs

```diff
@@ -123,23 +123,31 @@ public void ReplaceAccentColor(string name)
         /// <param name="newValue">The new entry value</param>
         /// <param name="parentDictionary">The root dictionary to start searching at. Null means using Application.Current.Resources</param>
         /// <returns>Weather the value was replaced (true) or not (false)</returns>
-        private static bool ReplaceEntry(object entryName, object newValue, ResourceDictionary parentDictionary = null)
+        private bool ReplaceEntry(object entryName, object newValue, ResourceDictionary parentDictionary = null ,bool iscolor = true)
         {
             if (parentDictionary == null)
                 parentDictionary = Application.Current.Resources;
 
             if (parentDictionary.Contains(entryName))
             {
-                parentDictionary[entryName] = newValue;
+                if (iscolor = true)
+                {
+                ColorAnimation animation;
+                animation = new ColorAnimation();
+                animation.From = (Color)parentDictionary[entryName]; //The old color
+                animation.To = (Color)newValue; //The new color
+                animation.Duration = new Duration(new TimeSpan(0,0,0,0,500)); //500ms animation
+                (parentDictionary[entryName] as Brush).BeginAnimation(SolidColorBrush.ColorProperty, animation);
+                }
+                else 
+                {
+                    parentDictionary[entryName] = newValue;
+                }
                 return true;
             }
 
             foreach (var dictionary in parentDictionary.MergedDictionaries)
-            {
-                if (ReplaceEntry(entryName, newValue, dictionary))
-                    return true;
-            }
-
+                return ReplaceEntry(entryName, newValue, dictionary);
```

Comment:
The 'if' there was necessary: In your new code, the ReplacyEntry method now returns false if the first merged dictionary fails to have the entry, never looking at the others.